float64 pterm
float64 iterm
float64 dterm
float64 signal
